### ISSUE TITLE
Page token for SPIsTranslation column

### ISSUE THREAD

**Issue by christouuu** [microsoft-search/pnp-modern-search#1814]
**Version used**
Ex: 4.2.3

**Describe the bug**
See additional context which may be important - especially we are in a bilingual FR/EN SharePoint communication site, in a news. 
Page token {Page._SPIsTranslation} is returning "Non" (browser in FR). 
Page token {Page._SPIsTranslation.value} is returning "undefined". 
No way to match the "SPIsTranslation" managed property with a page token. 

**To Reproduce**
1. Insert a search results webpart in a news of a SharePoint communication site. Make sure there are other news on the site.
2. Set the query like "Path:{Site.absoluteUrl} PromotedState:2 SPIsTranslation:{Page._SPIsTranslation}"
3. Apply and save the page properties

**Expected behavior**
Expected the _SPIsTranslation to have the value 0 or false so that it can be matched in a search query. 
OR the _SPIsTranslation.value to be usable in the query somehow as "SPIsTranslation:0" would work. 

**Screenshots**
![image](https://user-images.githubusercontent.com/16795711/153191158-14ab5092-48c0-4c59-af79-d46e00721b36.png)

**Desktop (please complete the following information):**
- Edge, Chrome

**Additional context**
I would like to use a search results webpart in a news page of a SharePoint communication site. 
I am on a multilingual site, but only 2 languages : FR/EN. 
My UI is in French. 
The query would be static as set fully in the properties, but with page tokens. 
I would like the query to match news of the same language. As I'm in a bilingual site, I wanted to use the SPIsTranslation managed property (=true/false). And match it to the page property _SPIsTranslation. The query being "SPIsTranslation:{Page._SPIsTranslation}"
RenderExtendedListFormData() is returning two values : 
- _SPIsTranslation = "Non"
- _SPIsTranslation.value = 0

So the query template that is sent in end is "SPIsTranslation:Non" which is not working as "Non" is literally in French and is neither "0" nor "false". 
I tried to put "SPIsTranslation:{Page._SPIsTranslation.value}" but that doesn't work - output of the query in the end is "SPIsTranslation:undefined". The .value is not considered as a subproperty of the complex type _SPIsTranslation, but _SPIsTranslation.value is considered as a value in full. 
![image](https://user-images.githubusercontent.com/16795711/153191158-14ab5092-48c0-4c59-af79-d46e00721b36.png)
I tried escaping something but didn't manage to make it work.

**Comment by wobba** [comment]
Thank you... table updated. At least my samples were correct :) Let me think about this some. _SPIsTranslation is a boolean property, letting you know if a page has translations or not.. nothing to do with the actual language.

**Comment by christouuu** [comment]
Samples definitely correct, again thanks for it - great great helping post :) 

In a parallel world, if a SPPageLanguage column/prop existed and always filled with a value, it would have been easy to match the search property to a filled page column value. Because my issue is resulting from SPTranslationLanguage being empty on the "master" language page - so unmatchable in a query (SPTranslationLanguage:"" will return nothing).

Why SPIsTranslation ? Well I'm trying to achieve an unperfect (wouldn't work if I had 3 languages) but working solution (because I only have 2).
In my case I have only 2 languages : French and English. 
SPIsTranslation always has a value : true/false. Hence if it's true, I know I'm in my translated page in English. If it's false, I know I'm in my French page. Both ways, SPIsTranslation:{Page._SPIsTranslation.value} would give me pages in the same language as the page I'm viewing : this is my need.

**Comment by wobba** [comment]
@christouuu You can also query against the managed properties `DetectedLanguage` and `languages` to get results - assuming there is enough text in the pages for a language to be detected.

**Comment by christouuu** [comment]
> @christouuu You can also query against the managed properties `DetectedLanguage` and `languages` to get results - assuming there is enough text in the pages for a language to be detected.

Yes, I tried but the detection is not accurate in my cases unfortunately. English pages seem detected as English in general. But French pages are sometimes detected as English

**Comment by wobba** [comment]
I guess the OOB expression would work for you, if we could provide the default site language, or the page's translated as a variable?

```
{?(SPTranslationLanguage:{Page.TranslatedLanguage} OR (NOT SPTranslatedLanguages:{Page.TranslatedLanguage}
AND NOT SPIsTranslation:true))
NOT}
(SPTranslationLanguage:{Site.Locale} OR (NOT SPTranslatedLanguages:{Site.Locale}
AND NOT SPIsTranslation:true))
```

I think the above would yield this for the default page:
`(SPTranslationLanguage:{Site.Locale} OR (NOT SPTranslatedLanguages:{Site.Locale}
AND NOT SPIsTranslation:true))`

and this for a translated page

`(SPTranslationLanguage:{Page.TranslatedLanguage} OR (NOT SPTranslatedLanguages:{Page.TranslatedLanguage}
AND NOT SPIsTranslation:true))
NOT
(SPTranslationLanguage:{Site.Locale} OR (NOT SPTranslatedLanguages:{Site.Locale}
AND NOT SPIsTranslation:true)`

**Comment by christouuu** [comment]
Ok I dug into the code and I think I found something. 

As I screenshoted, RenderExtendedListFormData API returns two simple string properties, that are lowercased in the getPageProperties() in TokenService.ts. 
"_spistranslation" is a simple string property. "_spistranslation.value" is considered as another simple string property, instead of being considered as complex "_spistranslation" property with a child property called "value". 
![image](https://user-images.githubusercontent.com/16795711/153191158-14ab5092-48c0-4c59-af79-d46e00721b36.png)

Then, to replace token values, and to retrieve the token value there is the ObjectHelper.ts with the byPath() static function. 
Because "_spistranslation.value" contains a dot, ObjectHelper.byPath() seems to compute as if .value is a child property when it's not. 
Therefore, byPath() is returning undefined for _spistranslation.value when it's retrieved by the API call. 

This may be something weird on the API side. 
But I think this may be solved by doing an exception in the replacePageTokens() for "_spistranslation.value" such as the if/else done to make a special treatment for taxonomy tokens (.Label and .TermID) ?
